src/slice-chop.h

```diff
@@ -3,8 +3,10 @@
 
 #include "vctrs-core.h"
 
+r_obj* vec_chop(r_obj* x, r_obj* indices, r_obj* sizes);
+r_obj* vec_chop_unsafe(r_obj*, r_obj* indices, r_obj* sizes);
```

Comment:
Moved `vec_chop()` from `<vctrs.h>` to `<slice-chop.h>`

And of course now `vec_chop()` is the version that does the index checks, and `vec_chop_unsafe()` is the performance variant that assumes the indices are already correct